osconfig_tests/test_suites/package_management/package_management.go

```diff
@@ -0,0 +1,187 @@
+var installOsConfigAgentGooGet = `c:\programdata\googet\googet.exe -noconfirm install -sources https://packages.cloud.google.com/yuck/repos/google-osconfig-agent-unstable google-osconfig-agent
+echo 'osconfig agent install done'`
+var installOsConfigAgentYumEL7 = `cat > /etc/yum.repos.d/google-osconfig-agent.repo <<EOM
+[google-osconfig-agent]
+name=Google OSConfig Agent Repository
+baseurl=https://packages.cloud.google.com/yum/repos/google-osconfig-agent-el7-unstable
+enabled=1
+gpgcheck=0
+repo_gpgcheck=1
+gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
+       https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+EOM
+yum -y install google-osconfig-agent
+echo 'osconfig agent install done'`
+
+type packageManagementTestSetup struct {
+	image       string
+	name        string
+	packageType []string
+	shortName   string
+	startup     *api.MetadataItems
+}
+
+// TestSuite is a PackageManagementTests test suite.
+func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junitxml.TestSuite, logger *log.Logger, testSuiteRegex, testCaseRegex *regexp.Regexp) {
+	defer tswg.Done()
+
+	if testSuiteRegex != nil && testSuiteRegex.MatchString(testSuiteName) {
+		return
+	}
+
+	testSuite := junitxml.NewTestSuite(testSuiteName)
+	defer testSuite.Finish(testSuites)
+
+	logger.Printf("Running TestSuite %q", testSuite.Name)
+
+	testSetup := []*packageManagementTestSetup{
+		// Debian images.
+		&packageManagementTestSetup{
+			image:       "projects/debian-cloud/global/images/family/debian-9",
+			packageType: []string{"deb"},
+			shortName:   "debian",
+			startup: &api.MetadataItems{
+				Key:   "startup-script",
+				Value: &installOsConfigAgentDeb,
+			},
+		},
+	}
+
+	var wg sync.WaitGroup
+	tests := make(chan *junitxml.TestCase)
+	for _, setup := range testSetup {
+		wg.Add(1)
+		go packageManagementTestCase(ctx, setup, tests, &wg, logger, testCaseRegex)
+	}
+
+	go func() {
+		wg.Wait()
+		close(tests)
+	}()
+
+	for ret := range tests {
+		testSuite.TestCase = append(testSuite.TestCase, ret)
+	}
+
+	logger.Printf("Finished TestSuite %q", testSuite.Name)
+}
+
+func runCreateOsConfigTest(ctx context.Context, testCase *junitxml.TestCase, logger *log.Logger) {
+	defer cleanupOsConfig(ctx, testCase, logger)
+
+	req := &osconfigpb.CreateOsConfigRequest{
+		Parent: "projects/compute-image-test-pool-001",
+		OsConfig: &osconfigpb.OsConfig{
+			Name: "foo",
+		},
+	}
+
+	logger.Printf("create osconfig request:\n%s\n\n", dump.Sprint(req))
+
+	res, err := osconfig_server.CreateOsConfig(ctx, logger, req)
+	if err != nil {
+		testCase.WriteFailure("error while creating osconfig:\n%s\n\n", err)
+	}
+
+	logger.Printf("CreateOsConfig response:\n%s\n\n", dump.Sprint(res))
+}
+
+// This function will cleanup all the osconfig created under project
+// Assumption is that this project is only used by this test application
+func cleanupOsConfig(ctx context.Context, testCase *junitxml.TestCase, logger *log.Logger) {
```

Comment:
Maybe use a key to filter on so this only cleans up configs for this test. Or have a cleanup wrapper around each osconfig you create or some sort of defer function.

Reply:
a wrapper function more sounds like what we want. will add one. thanks for the suggestion